backend/src/main/java/com/google/blackswan/mock/SimpleAnomalyGenerator.java

```diff
@@ -33,53 +33,52 @@
 import java.nio.charset.StandardCharsets;
 import java.io.ByteArrayInputStream;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import com.google.blackswan.mock.filesystem.*;
 
 /** Generate list of anomalies based on data in the csv file using average and threshold. */
 public class SimpleAnomalyGenerator implements AnomalyGenerator {
-  private static final String DATA_FILE_LOCATION = "/sample-ramen-data.csv";
   private static final int THRESHOLD = 13;
   private static final int NUM_POINTS = 5;
-  private static final String DEFAULT_METRIC = "interest";
-  private static final String DEFAULT_DIMENSION = "ramen";
 
   private final ImmutableList<Anomaly> anomalies;
+  private final DataInfo topic;
+  private final ImmutableMap<Timestamp, Integer> data;
 
-  /** 
-   * TODO: Let generators take in parameters of dimension/metric names. 
-   *       Can also take in THRESHOLD and NUM_POINTS as parameters.
-   */
-  public static SimpleAnomalyGenerator createGenerator() {
+  public static SimpleAnomalyGenerator createGenerator(DataInfo topic) {
```

Comment:
so the generator now depends on a topic (metric/dimension) to generate anomalies from?

Reply:
Yep, originally it was default to `dimension=ramen` and `metric=interest over time`, but now you can specify the topic you want the `SimpleAnomalyGenerator` to work with :)